types/proposal.go

```diff
@@ -23,36 +23,27 @@ var (
 // a so-called Proof-of-Lock (POL) round, as noted in the POLRound.
 // If POLRound >= 0, then BlockID corresponds to the block that is locked in POLRound.
 type Proposal struct {
-	Type          tmproto.SignedMsgType
-	Height        int64                   `json:"height"`
-	Round         int32                   `json:"round"`     // there can not be greater than 2_147_483_647 rounds
-	POLRound      int32                   `json:"pol_round"` // -1 if null.
-	BlockID       BlockID                 `json:"block_id"`
-	Timestamp     time.Time               `json:"timestamp"`
-	Signature     []byte                  `json:"signature"`
-	DAHeader      *DataAvailabilityHeader `json:"da_header"`
-	PartSetHeader PartSetHeader           `json:"part_set_header"`
+	Type      tmproto.SignedMsgType
+	Height    int64                   `json:"height"`
+	Round     int32                   `json:"round"`     // there can not be greater than 2_147_483_647 rounds
+	POLRound  int32                   `json:"pol_round"` // -1 if null.
+	BlockID   BlockID                 `json:"block_id"`
+	Timestamp time.Time               `json:"timestamp"`
+	Signature []byte                  `json:"signature"`
+	DAHeader  *DataAvailabilityHeader `json:"da_header"`
 }
 
 // NewProposal returns a new Proposal.
 // If there is no POLRound, polRound should be -1.
-func NewProposal(
-	height int64,
-	round int32,
-	polRound int32,
-	blockID BlockID,
-	daH *DataAvailabilityHeader,
-	psh PartSetHeader,
-) *Proposal {
+func NewProposal(height int64, round int32, polRound int32, blockID BlockID, daH *DataAvailabilityHeader) *Proposal {
 	return &Proposal{
-		Type:          tmproto.ProposalType,
-		Height:        height,
-		Round:         round,
-		BlockID:       blockID,
-		POLRound:      polRound,
-		Timestamp:     tmtime.Now(),
-		DAHeader:      daH,
-		PartSetHeader: psh,
+		Type:      tmproto.ProposalType,
+		Height:    height,
+		Round:     round,
+		BlockID:   blockID,
+		POLRound:  polRound,
+		Timestamp: tmtime.Now(),
+		DAHeader:  daH,
```

Comment:
Why does the DAHeader need to be part of the proposal again? Is that the case in the specs? cc @adlerjohn

Reply:
I was wondering the same thing. I defo don't think we need it. Reverting that might be slighty more complicated, as there have been quite a few changes since we merged #248 . We might want to revert that later or "manually" (not using `git revert`) later.

Reply:
The specs should be changed as needed to conform to the new Tendermint-as-black-box approach, so it's not really relevant in this time to chaos what the specs say. Since it's no longer necessary for validators to sample data, and since validators will in the happy path download the entire block data via parts, there's not much need anymore to include the DAH in the proposal. Note that the unhappy fallback case involves validator and other full nodes attempting to reconstruct the block by sampling individual shares until recovery is possible.

Reply:
OK, but there is still some nuance on what the validators should sing during consensus, right? 

Are you tracking the changes on the spec side for the tendermint-as-black-box approach? Or should we just build the thing first and update the specs later only?

Reply:
Since the the DAH won't be in the Proposal and not in the block, then I'm assuming that each validator/full node will recreate the DAH using the block data?

Reply:
> Are you tracking the changes on the spec side for the tendermint-as-black-box approach? Or should we just build the thing first and update the specs later only?

Build the thing first, then I'll do a pass after my vacation and make sure the specs and impl are aligned.

Reply:
> Since the the DAH won't be in the Proposal or in the block, then I'm assuming that each validator/full node will have to recreate the DAH using the block data?

Wait, isn't the DAH in the block, and gossiped as parts?

Reply:
> Wait, isn't the DAH in the block, and gossiped as parts?

Yes! but we're thinking of getting rid of that too I think. I can't find the comment in ADR 007, but it was removed from the list of minimal changes to tendermint because it was not strictly necessay. This reminds that we did not discuss that properly, and totally should, so I'll open an issue.

Reply:
Thanks Evan. IMO, there is no need to gossip the DAH as part of the block parts. The partsetheader already authenticates the Block and the receiving nodes can reconstruct the the DAH. But I might be missing sth here.